cg/meta/clean/clean_flow_cells.py

```diff
@@ -0,0 +1,136 @@
+"""An API that handles the cleaning of flow cells on Hasta."""
+import logging
+import time
+from pathlib import Path
+from typing import List
+
+from housekeeper.store.models import File
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import SequencingFileTag
+from cg.constants.time import TWENTY_ONE_DAYS_IN_SECONDS
+from cg.models.demultiplex.flow_cell import FlowCellDirectoryData
+from cg.store import Store
+from cg.store.models import Flowcell, SampleLaneSequencingMetrics
+from cg.utils.files import get_creation_time_stamp, remove_directory_and_contents
+
+LOG = logging.getLogger(__name__)
+
+
+class CleanFlowCellAPI:
+    """
+            Handles the cleaning of flow cells in the flow_cells and demultiplexed_runs directories.
+    Requirements for cleaning:
+            Flow cell is older than 21 days
+            Flow cell is backed up
+            Flow cell is in StatusDB
+            Flow cell has sequencing metrics in StatusDB
+            Flow cell has fastq files in Housekeeper
+            Flow cell has SPRING files in Housekeeper
+            Flow cell has a sample sheet in Housekeeper
+    """
+
+    def __init__(
+        self, status_db: Store, housekeeper_api: HousekeeperAPI, flow_cell_path: Path, dry_run: bool
+    ):
+        self.status_db: Store = status_db
+        self.hk_api: HousekeeperAPI = housekeeper_api
+        self.flow_cell = FlowCellDirectoryData(flow_cell_path=flow_cell_path)
+        self.current_time = time.time()
+        self.dry_run: bool = dry_run
+
+    def delete_flow_cell_directory(self):
+        """Delete the flow cell directory if it fulfills all requirements."""
+        is_error_raised: bool = False
+        try:
+            if self.can_flow_cell_directory_be_deleted():
+                if self.dry_run:
+                    LOG.debug(f"Dry run: Would have removed: {self.flow_cell.path}")
+                    return
+                remove_directory_and_contents(self.flow_cell.path)
+                return is_error_raised
+        except Exception as error:
+            is_error_raised = True
+            LOG.error(f"Flow cell with path {self.flow_cell.path} not removed: {str(error)}")
+            return is_error_raised
+
+    def can_flow_cell_directory_be_deleted(self) -> bool:
+        """Determine whether a flow cell directory can be deleted."""
+        return all(
+            [
+                self.is_directory_older_than_days_old(),
+                self.is_flow_cell_in_statusdb(),
+                self.is_flow_cell_backed_up(),
+                self.has_sequencing_metrics(),
+                self.has_fastq_files_in_housekeeper(),
+                self.has_spring_files_in_housekeeper(),
+                self.has_sample_sheet_in_housekeeper(),
+            ]
+        )
+
+    def is_directory_older_than_days_old(self) -> bool:
```

Comment:
Seems like s general method if one can supply the threshold. Maybe something that can be DRY in `cg/utils`?

Reply:
Yes, great idea! I will move it there.